katello/katello-backup

```diff
@@ -106,6 +107,20 @@ def compress_files
   Process.wait(mongo)
 end
 
+def generate_metadata
+  metadata = File.new 'metadata.yaml', 'w'
+  os_version = `cat /etc/redhat-release`.chomp
+  plugin_list = `foreman-rake plugin:list | grep 'Foreman plugin: '`.lines
+  plugins = []
+  plugin_list.each do |line|
+    plugin = line.split
+    plugins << "#{plugin[2].chop}-#{plugin[3].chop}"
+  end
+  rpms = `rpm -qa`.split("\n")
+  system_facts = {os_version: os_version, plugin_list: plugins, rpms: rpms}
+  metadata.puts system_facts.to_yaml
```

Comment:
make sure to close the file:
`metadata.close`.

You might also split out the plugin list fetching into its own method.